### PR TITLE
Fix/create balsamic scout config root

### PR DESCRIPTION
**Pull request by karlnyr**
### Fixed

- Config builder standalone for balsamic

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/upload/scout.py [hunk 1]
```diff
@@ -56,17 +59,26 @@ def scout(context, re_upload: bool, print_console: bool, case_id: str):
...
     LOG.info("Fetching family object")
     case_obj: Family = status_db.family(case_id)
-    LOG.info("Create load config")
+
     if not case_obj.analyses:
         LOG.warning("Could not find analyses for %s", case_id)
         raise click.Abort
+
+    if case_obj.data_analysis == Pipeline.BALSAMIC:
+        LOG.info(f"Found BALSAMIC analysis for {case_id}, using BALSAMIC upload API")
```
Comment (henrikstranneheim) [inline cg/cli/upload/scout.py:72]:
```suggestion
        LOG.info(f"Found {Pipeline.BALSAMIC} analysis for {case_id}, using {Pipeline.BALSAMIC} upload API")
```

File: cg/cli/upload/scout.py [hunk 2]
```diff
@@ -56,17 +59,26 @@ def scout(context, re_upload: bool, print_console: bool, case_id: str):
...
+        context.meta_apis["analysis_api"] = BalsamicAnalysisAPI(context)
+    elif case_obj.data_analysis == Pipeline.BALSAMIC_UMI:
+        LOG.info(f"Found BALSAMIC UMI analysis for {case_id}, using BALSAMIC UMI upload API")
```
Comment (henrikstranneheim) [inline cg/cli/upload/scout.py:75]:
```suggestion
        LOG.info(f"Found {Pipeline.BALSAMIC_UMI} analysis for {case_id}, using {Pipeline.BALSAMIC_UMI} upload API")
```

File: cg/cli/upload/scout.py [hunk 3]
```diff
@@ -56,17 +59,26 @@ def scout(context, re_upload: bool, print_console: bool, case_id: str):
...
+        context.meta_apis["analysis_api"] = BalsamicUmiAnalysisAPI(context)
```
Comment (henrikstranneheim) [inline cg/cli/upload/scout.py:76]: Add type hint

File: tests/cli/upload/test_cli_scout.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+def test_create_scout_load_config(
+    caplog,
+    cli_runner: CliRunner,
+    cg_context: CGConfig,
+    case_id: str,
+    helpers: StoreHelpers,
+    upload_scout_api: MockScoutUploadApi,
+):
+    """Test to create a scout load config for a BALSAMIC case"""
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_scout.py:24]:
```suggestion
    """Test to create a scout load config for a BALSAMIC case."""
```

File: tests/cli/upload/test_cli_scout.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    caplog.set_level(logging.DEBUG)
+    status_db: Store = cg_context.status_db
+    cg_context.meta_apis["upload_api"] = BalsamicUploadAPI(cg_context)
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_scout.py:27]: Add type hints also in tests to make it easier to understand the requirements

File: tests/cli/upload/test_cli_scout.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    cg_context.meta_apis["upload_api"].scout_upload_api = upload_scout_api
+    # GIVEN a case with a balsamic analysis
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_scout.py:29]:
```suggestion

    # GIVEN a case with a balsamic analysis
```

File: tests/cli/upload/test_cli_scout.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    case: Family = helpers.ensure_case(
+        store=status_db, data_analysis=Pipeline.BALSAMIC, case_id=case_id
+    )
+    helpers.add_analysis(store=status_db, pipeline=Pipeline.BALSAMIC, case=case)
+    # GIVEN a BalsamicUploadAPI with an AnalysisAPI with a root
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_scout.py:34]:
```suggestion

    # GIVEN a BalsamicUploadAPI with an AnalysisAPI with a root
```

File: tests/cli/upload/test_cli_scout.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    upload_api: BalsamicUploadAPI = BalsamicUploadAPI(cg_context)
+    # WHEN creating the scout load config
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_scout.py:36]:
```suggestion

    # WHEN creating the scout load config
```

File: tests/cli/upload/test_cli_scout.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    result = cli_runner.invoke(create_scout_load_config, ["--print", case_id], obj=cg_context)
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_scout.py:37]: Should we not change `--print` to `--dry-run`?

File: tests/cli/upload/test_cli_scout.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    # THEN assert that the correct root is set
+    assert result.exit_code == 0
+    assert upload_api.analysis_api.root in caplog.text
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_scout.py:40]:
```suggestion
    # THEN exit successfully
    assert result.exit_code == EXIT_SUCCESS
    
    # THEN assert that the BALSAMIC API root is set
    assert upload_api.analysis_api.root in caplog.text
```

**Review by henrikstranneheim — APPROVED** [review]
Great that you added a test 👍 . There  also seems to be an empty dir here `tests/cli/upload/scout` that we can clean.

**Comment by karlnyr** [comment]
Test performed on awareroughy, the path is now correctly set to the appropriate analysis type.

**Comment by karlnyr** [comment]
Installed in prod:

```Command
From https://github.com/Clinical-Genomics/cg
   e3038d0..68004c8  master     -> origin/master
   7479861..2602d34  add_file_tags_to_protect_in_Housekeeper_for_balsamic_analysis -> origin/add_file_tags_to_protect_in_Housekeeper_for_balsamic_analysis
 * [new branch]      extend_basehandler -> origin/extend_basehandler
 * [new branch]      fix-start-downsampled-cases -> origin/fix-start-downsampled-cases
 * [new branch]      move-filters-into-separate-directory -> origin/move-filters-into-separate-directory
 * [new branch]      refactor-organism-sql -> origin/refactor-organism-sql
 * [new branch]      refactor-panel-sql -> origin/refactor-panel-sql
 * [new branch]      refactor-update-case-command -> origin/refactor-update-case-command
 * [new branch]      refactor-user-sql -> origin/refactor-user-sql
   7fb2c7a..afde217  refactor_customer_sql -> origin/refactor_customer_sql
   d612b3d..0b5056c  refactor_invoice_sql -> origin/refactor_invoice_sql
   5f9f8ef..18ad800  remove/balsamic-concatenation -> origin/remove/balsamic-concatenation
   b58f4f5..0a3f34a  upload-mip-rna-to-collaborators -> origin/upload-mip-rna-to-collaborators
 * [new tag]         v26.7.3    -> v26.7.3
From https://github.com/Clinical-Genomics/cg
 * [new tag]         v26.7.2    -> v26.7.2
Already on 'master'
Your branch is behind 'origin/master' by 4 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Updating e3038d0..68004c8
Fast-forward
 .bumpversion.cfg                   |  2 +-
 Dockerfile                         |  2 --
 cg/__init__.py                     |  2 +-
 cg/cli/upload/base.py              |  2 +-
 cg/cli/upload/scout.py             | 36 +++++++++++++++++++++++++++++++-----
 cg/meta/upload/upload_api.py       |  2 +-
 cg/server/app.py                   |  4 +---
 cg/server/config.py                |  2 --
 cg/server/ext.py                   |  2 +-
 setup.py                           |  2 +-
 tests/cli/upload/test_cli_scout.py | 61 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 tests/mocks/scout.py               | 40 +++++++++++++++++++++++++++++++++++++++-
 tests/server/conftest.py           |  1 -
 13 files changed, 138 insertions(+), 20 deletions(-)
 create mode 100644 tests/cli/upload/test_cli_scout.py
/home/hiseq.clinical
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical
done.
Log deploy... done.
cg, version 26.7.3
```

**Comment by karlnyr** [comment]
I forgot to add the (patch) when merging, there will be no new version of this deployed to the vms nor a version bump. Should be fine and fixed once someone else does a bump.

**Comment by Vince-janv** [comment]
> I forgot to add the (patch) when merging, there will be no new version of this deployed to the vms nor a version bump. Should be fine and fixed once someone else does a bump.

It defaults to a patch so you're fine 🌟

**Pull request merged**